csrc/fusion_segmenter.h

```diff
@@ -236,6 +242,12 @@ class SegmentedGroup {
 
   //! SegmentedFusion this group belongs to
   SegmentedFusion* segmented_fusion_;
+
+  //! The cloned segmented fusion
+  std::unique_ptr<Fusion> cloned_fusion_;
```

Comment:
Cleanup suggestions: can we move this and `original_inputs_in_cloned_fusion_` to SegmentedFusion? SegmentedGroup has been a minimal representation for a subgraph view into the original fusion. Therefore, SegmentedFusion owns a list of SegmentedGroups but a SegmentedGroup only refers to the original fusion. With this change as is, a SegmentedGroup owns also a cloned fusion, a bit too complicated for what it's designed for.

Reply:
I see your point about keeping `SegmentedGroup` a simple class.  However, it is more cumbersome to serialize. 

Your suggestion would mean we'd have a `std::unordered_map<SegmentedGroup*, ClonedFusionData>` in `SegmentedFusion`. For serialization, we'd separate the keys and values into separate lists because maps aren't serializable. The pointer values would be mapped to an integer id that we'd check for during deserialization. It is much easier to have data contained in the class it is associated with rather than mapping through pointers.

Reply:
> making multiple fusions for the same SegmentedGroup

Wondering where's *making multiple fusions for the same segmentedGroup* happening? Weren't there only one SegmentedGroup instance per group and they only make one copy of fusion inside?

Reply:
There are SegmentedFusion::makeFusion, which always creates a fusion, and SegmentedGroup::getFusion, which yes tries to reuse the cached.